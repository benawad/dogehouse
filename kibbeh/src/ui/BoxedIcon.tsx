import React, { ReactElement } from "react";

const colorMap = {
  "700": "bg-primary-700",
  "800": "bg-primary-800",
};

export interface BoxedIconProps
  extends React.ComponentPropsWithoutRef<"button"> {
  circle?: boolean;
  transition?: boolean;
  hover?: boolean;
  color?: keyof typeof colorMap;
}

export const BoxedIcon: React.FC<BoxedIconProps> = ({
  color = "700",
  children,
  className = "",
  circle = false,
  transition = false,
  hover = false,
  ...props
}) => {
  return (
    <button
      className={`flex ${colorMap[color]} ${
        transition ? `transition duration-200 ease-in-out` : ``
      } ${
        hover ? `` : `hover:bg-primary-600`
      } h-6 w-6 cursor-pointer justify-center items-center ${
        circle ? `rounded-full` : `rounded`
      } ${className.includes("text-button") ? "" : "text-primary-100"}
        ${className}`}
      data-testid="boxed-icon"
      {...props}
    >
      {children}
    </button>
  );
};
