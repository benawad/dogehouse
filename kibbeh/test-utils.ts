import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import "mutationobserver-shim";

const customRender = (ui: any, options = {}) =>
  render(ui, {
    ...options,
  });

export * from "@testing-library/react";

export { customRender as render };
