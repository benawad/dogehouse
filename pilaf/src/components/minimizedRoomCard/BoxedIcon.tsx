import React from "react";
import {
  Image,
  ImageSourcePropType,
  StyleProp,
  StyleSheet,
  TouchableOpacity,
  ViewStyle,
} from "react-native";
import { colors, radius } from "../../constants/dogeStyle";

interface BoxedIconProps {
  style?: StyleProp<ViewStyle>;
  image: ImageSourcePropType;
  imageColor?: string;
  onPress: () => void;
}

export const BoxedIcon: React.FC<BoxedIconProps> = ({
  style,
  image,
  imageColor = colors.text,
  onPress,
}) => {
  return (
    <TouchableOpacity style={[styles.container, style]} onPress={onPress}>
      <Image source={image} style={{ tintColor: imageColor }} />
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    justifyContent: "center",
    alignItems: "center",
    width: 40,
    height: 40,
    backgroundColor: colors.primary700,
    borderRadius: radius.m,
  },
});
