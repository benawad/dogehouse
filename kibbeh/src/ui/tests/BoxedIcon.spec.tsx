import React from "react";

import { render } from "../../../test-utils";
import { BoxedIcon } from "../BoxedIcon";

describe("BoxedIcon", () => {
  describe("BoxedIcon", () => {
    it("should render correctly", () => {
      const { getByTestId, getByText } = render(<BoxedIcon>Child</BoxedIcon>);
      const component = getByTestId("boxed-icon");
      const child = getByText("Child");

      expect(component).toBeVisible();
      expect(child).toBeVisible();
    });

    it("should match snapshot", () => {
      const { getByTestId } = render(<BoxedIcon />);
      const component = getByTestId("boxed-icon");

      expect(component).toMatchSnapshot();
    });
  });
});
