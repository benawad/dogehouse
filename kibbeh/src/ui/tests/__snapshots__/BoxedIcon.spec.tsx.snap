// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxedIcon BoxedIcon should match snapshot 1`] = `
<button
  class="flex bg-primary-700  hover:bg-primary-600 h-6 w-6 cursor-pointer justify-center items-center rounded text-primary-100
        "
  data-testid="boxed-icon"
/>
`;
